tasks/package.js

```diff
@@ -13,35 +13,110 @@ module.exports = function(grunt) {
     var path = require('path');
     grunt.loadNpmTasks('grunt-contrib-copy');
 
-    var config = grunt.config.get('config.packages'),
+    var config = grunt.config.get('config.packages.default'),
       srcFiles = ['**', '!**/.gitkeep'].concat((config && config.srcFiles && config.srcFiles.length) ? config.srcFiles : '**'),
       projFiles = (config && config.projFiles && config.projFiles.length) ? config.projFiles : [];
 
     var destPath = grunt.config.get('config.buildPaths.package') + '/package';
     var tasks = []
 
-    grunt.config('copy.package', {
-      files: [
-        {
-          expand: true,
-          cwd: '<%= config.buildPaths.html %>',
-          src: srcFiles,
-          dest: destPath + (grunt.config.get('config.packages.dest.docroot') || ''),
-          dot: true
-        },
-        {
+    // Figure out if we are building a vendor-specific package.
+    var buildTarget = grunt.option('target') || process.env.GRUNT_PACKAGE_TARGET;
+    if (buildTarget && isValidTarget(buildTarget)) {
+      grunt.config('config.buildTarget', grunt.option('target'))
+    }
+    else {
+      grunt.config('config.buildTarget', 'default');
+    }
+    grunt.log.writeln('Building package for ' + grunt.config.get('config.buildTarget'));
+
+    var validTarget = grunt.config.get('config.buildTarget');
+
+    // Build the specified vendor target, or the default.
+    switch (validTarget) {
```

Comment:
Agreed. The wrangling was in the interest of supporting the segmented repo structure mentioned there. More habit here in validating user input than anything else. If the standard is "make acquia like `build/html`, then this is all cruft.